pandas/computation/align.py

```diff
@@ -172,12 +172,8 @@ def _reconstruct_object(typ, obj, axes, dtype):
         ret_value = res_t.type(obj)
     else:
         ret_value = typ(obj).astype(res_t)
+        if len(obj.shape) == 1 and len(obj) == 1:
```

Comment:
ok, then ok (but maybe add a comment on what you are checking for here). otherwise lgtm. merge when ready.